nsot/models.py

```diff
@@ -1396,8 +1396,43 @@ def set_addresses(self, addresses, overwrite=False, partial=False):
 
         self.clean_addresses()
 
+    def get_ancestors(self):
+        """Recursively get all parents of an interface"""
+        p = self.parent
+        ancestors = []
+        while p is not None:
+            ancestors.append(p)
+            p = p.parent
+        return ancestors
+
+    def get_children(self):
+        """Return the immediate children of an interface"""
+        return Interface.objects.filter(parent=self)
+
+    def get_descendants(self):
+        """Recursively return all the children of an interface"""
+        s = list(self.get_children())
+        children = []
```

Comment:
Just a comment: Similarly to ancestors, returning a list may be problematic if we ever need to filter (or chain filters) on descendant interfaces.